tasks/Task.cpp

```diff
@@ -0,0 +1,167 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "Task.hpp"
+#include "../Mapping.hpp"
+
+#include <base-logging/Logging.hpp>
+
+#include <seasocks/PrintfLogger.h>
+#include <seasocks/WebSocket.h>
+#include <seasocks/Server.h>
+
+#include <json/json.h>
+
+#include <string.h>
+#include <iostream>
+
+using namespace controldev_websocket;
+using namespace controldev;
+using namespace seasocks;
+
+struct controldev_websocket::JoystickHandler : WebSocket::Handler {
+    WebSocket *connection = nullptr;
+    Task *task = nullptr;
+    Json::Value msg;
+    Json::FastWriter fast;
+    void onConnect(WebSocket *socket) override{
+        if (connection != nullptr){
+            connection->close();
+        }
+        connection = socket;
+    }
+    void onData(WebSocket *socket, const char *data) override{
+        msg["result"] = false;
+        if (connection == socket){
+            ++task->received;
+            msg["result"] = Json::Value(task->updateRawCommand(data));
+            if (msg["result"].asBool() && task->controlling){
+                task->raw_cmd_obj.time = base::Time::now();
+                task->_raw_command.write(task->raw_cmd_obj);
+            } else {
+                ++task->errors;
+            }
+        }
+        connection->send(fast.write(msg));
+    }
+    void onDisconnect(WebSocket *socket) override{
+        if (connection == socket){
+            connection = nullptr;
+        }
+    }
+};
+
+struct controldev_websocket::WrapperJSON{
+    Json::Value jdata;
+    Json::CharReaderBuilder rbuilder;
+    std::unique_ptr<Json::CharReader> const reader;
+
+    WrapperJSON() : reader(rbuilder.newCharReader()) {
+
+    }
+
+    double getValue(const std::string &field, int index){
+        return jdata[field][index].asDouble();
+    }
+};
+
+bool Task::updateRawCommand(const char *data){
+    std::string errs;
+    // If an error occurs trying to get the parameters should not ignore it.
+    auto axis = _axis_map.get();
+    auto button = _button_map.get();
+
+    if (!wrapper->reader->parse(data, data+std::strlen(data), &wrapper->jdata, &errs)){
+        LOG_ERROR_S << "Failed parsing the message, got error: " << errs << std::endl;
+        return false;
+    }
+    if (!wrapper->jdata.isMember("status") ||
+        !wrapper->jdata.isMember(Mapping::mapFieldName(Mapping::Type::Axis)) ||
+        !wrapper->jdata.isMember(Mapping::mapFieldName(Mapping::Type::Button))) {
```

Comment:
This is too much. Just put the strings directly.

Reply:
But is safer. Really prefer that?

Reply:
I see your point. I would feel a lot less weird if you would move mapFieldName to this file as I requested. It really has nothing to do anywhere else.

`mapFieldName`'s only goal is to be compatible with whatever JSON is processed by this task. It must be defined here.

Reply:
I moved yesterday @doudou